backend/functions/lib/use-cases/acms-orders/acms-orders.ts

```diff
@@ -0,0 +1,193 @@
+import { ApplicationContext } from '../../adapters/types/basic';
+import Factory from '../../factory';
+import { ConsolidationFrom, ConsolidationTo } from '../../../../../common/src/cams/events';
+import { ConsolidationType } from '../../../../../common/src/cams/orders';
+import { CaseSummary } from '../../../../../common/src/cams/cases';
+import { CaseConsolidationHistory } from '../../../../../common/src/cams/history';
+import { ACMS_SYSTEM_USER_REFERENCE } from '../../../../../common/src/cams/auditable';
+import { getCamsError } from '../../common-errors/error-utilities';
+
+const MODULE_NAME = 'ACMS_ORDERS_USE_CASE';
+
+export type Bounds = {
+  divisionCodes: string[];
+  chapters: string[];
+};
+
+export type Predicate = {
+  divisionCode: string;
+  chapter: string;
+};
+
+export type PredicateAndPage = Predicate & {
+  pageNumber: number;
+};
+
+export type AcmsConsolidationChildCase = {
+  caseId: string;
+  consolidationType: string;
+  consolidationDate: string;
+};
+
+export type AcmsConsolidation = {
+  leadCaseId: string;
+  childCases: AcmsConsolidationChildCase[];
+};
+
+export type AcmsConsolidationReport = {
+  leadCaseId: string;
+  success: boolean;
+};
+
+export class AcmsOrders {
+  public async getPageCount(context: ApplicationContext, predicate: Predicate): Promise<number> {
+    const gateway = Factory.getAcmsGateway(context);
+    return gateway.getPageCount(context, predicate);
+  }
+
+  public async getLeadCaseIds(
+    context: ApplicationContext,
+    predicateAndPage: PredicateAndPage,
+  ): Promise<string[]> {
+    const gateway = Factory.getAcmsGateway(context);
+    return gateway.getLeadCaseIds(context, predicateAndPage);
+  }
+
+  public async migrateConsolidation(
+    context: ApplicationContext,
+    leadCaseId: string,
+  ): Promise<AcmsConsolidationReport> {
+    // TODO: Add child case count to the report??
+    const report: AcmsConsolidationReport = { leadCaseId, success: true };
+    try {
+      const casesRepo = Factory.getCasesRepository(context);
+      const dxtr = Factory.getCasesGateway(context);
+      const acms = Factory.getAcmsGateway(context);
+
+      const basics = await acms.getConsolidationDetails(context, leadCaseId);
+      const leadCase = await dxtr.getCaseSummary(context, leadCaseId);
+
+      // NOTE! Azure suggests that all work be IDEMPOTENT because activities run _at least once_.
+      // Check if exported child cases have already been migrated.
```

Comment:
For the idempotent note, I'd like to add it to a JSDOC. The other one seems like a TODO?